### PR TITLE
MAT-9181 Spliting the buildTranslator logic for STU7

### PR DESCRIPTION
**Pull request by RohitKandimalla**
## CQL to ELM Translation Service PR

Jira Ticket: [MAT-0000](https://jira.cms.gov/browse/MAT-0000)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/mat/cql_elm_translation/service/CqlTooling.java [hunk 1]
```diff
@@ -109,11 +109,16 @@ protected CqlTranslator runTranslator(
...
     UsingProperties usingProperties = cqlTextParser.getUsing();
-    return TranslationResource.getInstance(
-            usingProperties != null
-                && ("FHIR".equals(usingProperties.getLibraryType())
-                    || "QICore".equals(usingProperties.getLibraryType())))
-        .buildTranslator(requestData);
+    boolean isFhir =
+        usingProperties != null
+            && ("FHIR".equals(usingProperties.getLibraryType())
+                || "QICore".equals(usingProperties.getLibraryType()));
```
Comment (nmorasb) [inline src/main/java/gov/cms/mat/cql_elm_translation/service/CqlTooling.java]: Can we add a check for USCore as well? With 7.0.0 we expect authors will have multiple using statements, and it's valid to have both `using QICore version '7.0.0'` and `using USCore version '7.0.0'` in any order.

File: src/main/java/gov/cms/mat/cql_elm_translation/service/CqlTooling.java [hunk 2]
```diff
@@ -109,11 +109,16 @@ protected CqlTranslator runTranslator(
...
+    // for now, we are treating QICore v7.0.0 the same as FHIR
+    if (isFhir && usingProperties.getVersion().equalsIgnoreCase("7.0.0")) {
```
Comment (nmorasb) [inline src/main/java/gov/cms/mat/cql_elm_translation/service/CqlTooling.java]: also, maybe this check can just see if the version is _at minimum_ `7.0.0` (maybe something like starts with `7.`)? I think 7.0.1 has already been released, and whenever we go to support STU8 it will also need to be treated the same way as STU7 (using the NpmModelInfoProvider)

Comment (RohitKandimalla) [reply src/main/java/gov/cms/mat/cql_elm_translation/service/CqlTooling.java]: Updated it to check for any versions that above v7

File: src/main/java/gov/cms/mat/cql_elm_translation/service/CqlTooling.java (hunk 1 above)
Comment (nmorasb) [reply src/main/java/gov/cms/mat/cql_elm_translation/service/CqlTooling.java]: Aren't we still missing the check for USCore?
So for example, if someone has CQL:

```
library NewMeasureLib

using USCore version '7.0.0'
using QICore version '7.0.0'
...
```
then this won't work properly

Comment (RohitKandimalla) [reply src/main/java/gov/cms/mat/cql_elm_translation/service/CqlTooling.java]: Added USCore to the clause

**Review by nmorasb — APPROVED** [review]
